text/0000-panic-in-drop.md

```diff
@@ -275,6 +275,16 @@ This mirrors the behavior of C++ where a throwing destructor can be opted into u
 
 Unfortunately this doesn't really fit Rust's safety model: such a type would have to be unsafe to create and use since it would be easy to invoke UB by using this type with an API that doesn't support unwinding drops.
 
+## Allow individual `Drop` impls to opt-out of unwinding
+
+This is a reverse of the previous point: the default behavior of drops could stay as it is, but individual `Drop` impls could opt-in to aborting if an unwind escapes it.
+
+However this doesn't help generic code such as collections which still need to be able to handle user-defined types which do abort on drop.
+
+## Only abort for unwinds out of implicit drops
+
+This would still allow unwinds to escape calls to `drop_in_place`. Abort guards would be generated for implicit drops at the end of a scope.
```

Comment:
I quite like this idea, why it's not good?

Reply:
It doesn't actually cover all the cases. For example it would not have caught [this](https://github.com/rust-lang/hashbrown/pull/348) bug in hashbrown.

Reply:
But the footgun in that code also applies to things like `+`, `>`, `.`, `{}`-formatting _etc._, so it kind of seems like an orthogonal concern

Reply:
In this case, that code is already handling the case where `T: Clone` panics, and forgot to handle the case where `drop_in_place::<T>` panics.

Reply:
Would it help having another function called `drop_in_place_may_panic`? That way `unsafe` code which actually put in the work to handle panics in drops correctly would be able to explicitly call it. `drop_in_place` would abort on panic.

Reply:
@danielhenrymantilla I think it's a very good objection. Got and idea: https://github.com/rust-lang/rust-clippy/issues/9200 - could be a step towards improving the situation.

Reply:
These other operations aren't quite as bad, since you need specific `T: Add` or `T: Deref` bounds before they're available. Every value of a generic type (except if you require `T: Copy`), and every trait object, can call an arbitrary `drop_in_place()` impl.